src/js/letters/containers/AddressSection.jsx

```diff
@@ -8,26 +8,63 @@ import {
   isDomesticAddress,
   isMilitaryAddress,
   isInternationalAddress,
-  invalidAddressProperty,
   addressUpdateUnavailable
 } from '../utils/helpers.jsx';
-import { updateAddress, saveAddress } from '../actions/letters';
+import { saveAddress } from '../actions/letters';
 import Address from '../components/Address';
+import InvalidAddress from '../components/InvalidAddress';
+import AddressContent from '../components/AddressContent';
 
 export class AddressSection extends React.Component {
   constructor() {
     super();
-    this.state = { isEditingAddress: false };
+    this.state = {
+      hasLoadedAddress: false,
+      isEditingAddress: false,
+      editableAddress: {},
```

Comment:
Given that the `savedAddress` _may_ come in before this component mounts, can we do something like this:

```js
constructor(props) {
  super(props);
  this.state = {
    hasLoadedAddress: false,
    isEditingAddress: false,
    editableAddress: props.savedAddress || {}
  }

  // If the request has already come in, make sure we know that we've loaded it already
  if (Object.keys(this.state.editableAddress) !== 0) {
    this.state.hasLoadedAddress = true;
  }
}
```

I don't know that it ever _will_ come in before this component mounts, but I just like to play it safe when it comes to asynchronous whatnots. This is maybe overkill. :man_shrugging:

Reply:
Yep, I like this! We'll provide better UX at least some of the time

Reply:
Actually, React seems to not like relying on an undefined prop in the constructor. I could put in some more logic to guard against this but not sure it's worth it.

Reply:
Doh, that's what happens when you don't pass props to the constructor...fixed